playground/Common/ArrayMaze.cs

```diff
@@ -0,0 +1,51 @@
+using System.Numerics;
+
+namespace Common;
+
+public class ArrayMaze : IMaze
```

Comment:
Do we really need the `Array` prefix?

Reply:
We may have different IMaze implementations, so it's fine to name interface Maze, but we might use an array of arrays of bools or some other method to store maze, so this name makes room for that.